File: 프로그래머스/lv0/120840. 구슬을 나누는 경우의 수/구슬을 나누는 경우의 수.py
# 조합 문제인 듯

def factorial(n):
    if n == 0 or n == 1:
        return 1
    return factorial(n-1)*n

def solution(balls, share):
    a = factorial(balls)//factorial(share)//factorial(balls-share)
    
    return a

File: 프로그래머스/0/120840. 구슬을 나누는 경우의 수/구슬을 나누는 경우의 수.java
import java.math.BigInteger;


class Solution {
    public BigInteger solution(int balls, int share) {
        return factorial(balls).divide(factorial(share)).divide(factorial(balls - share));
    }

    public BigInteger factorial(int n) {
        if (n == 0 || n == 1) {
            return BigInteger.ONE;
        }

        return factorial(n - 1).multiply(BigInteger.valueOf(n));
    }
}
